custom/icds_reports/utils/aggregation.py

```diff
@@ -1024,3 +1025,293 @@ def aggregation_query(self):
+            ('wasting_normal',
+                "SUM(CASE WHEN ucr.wasting_normal = 1 AND ucr.nutrition_status_weighed = 1 "
+                "AND ucr.height_measured_in_month = 1 THEN 1 ELSE 0 END)"),
+            ('stunting_normal',
+                "SUM(CASE WHEN ucr.stunting_normal = 1 AND ucr.height_measured_in_month = 1 "
+                "THEN 1 ELSE 0 END)"),
+            ('valid_all_registered_in_month', "SUM(chm.valid_all_registered_in_month)"),
+            ('ebf_no_info_recorded', "SUM(chm.ebf_no_info_recorded)"),
+            ('weighed_and_height_measured_in_month',
+                "SUM(CASE WHEN chm.nutrition_status_weighed = 1 AND ucr.height_measured_in_month = 1 "
+                "THEN 1 ELSE 0 END)"),
+            ('weighed_and_born_in_month',
+                "SUM(CASE WHEN (chm.born_in_month = 1 AND (chm.nutrition_status_weighed = 1 "
+                "OR chm.low_birth_weight_born_in_month = 1)) THEN 1 ELSE 0 END)"),
+            ('zscore_grading_hfa_normal',
+                "SUM(CASE WHEN chm.zscore_grading_hfa_recorded_in_month = 1 AND "
+                "chm.zscore_grading_hfa = 3 THEN 1 ELSE 0 END)"),
+            ('zscore_grading_hfa_moderate',
+                "SUM(CASE WHEN chm.zscore_grading_hfa_recorded_in_month = 1 AND "
+                "chm.zscore_grading_hfa = 2 THEN 1 ELSE 0 END)"),
+            ('zscore_grading_hfa_severe',
+                "SUM(CASE WHEN chm.zscore_grading_hfa_recorded_in_month = 1 AND "
+                "chm.zscore_grading_hfa = 1 THEN 1 ELSE 0 END)"),
+            ('wasting_normal_v2',
+                "SUM(CASE WHEN chm.zscore_grading_wfh_recorded_in_month = 1 AND chm.zscore_grading_wfh = 3 THEN 1 "
+                "WHEN chm.muac_grading_recorded_in_month = 1 AND chm.muac_grading = 3 THEN 1 "
+                "ELSE 0 END)"),
+            ('wasting_moderate_v2',
+                "SUM(CASE WHEN chm.zscore_grading_wfh_recorded_in_month = 1 AND chm.zscore_grading_wfh = 2 THEN 1 "
+                "WHEN chm.muac_grading_recorded_in_month = 1 AND chm.muac_grading = 2 THEN 1 "
+                "ELSE 0 END)"),
+            ('wasting_severe_v2',
+                "SUM(CASE WHEN chm.zscore_grading_wfh_recorded_in_month = 1 AND chm.zscore_grading_wfh = 1 THEN 1 "
+                "WHEN chm.muac_grading_recorded_in_month = 1 AND chm.muac_grading = 1 THEN 1 "
+                "ELSE 0 END)"),
+            ('zscore_grading_hfa_recorded_in_month', "SUM(chm.zscore_grading_hfa_recorded_in_month)"),
+            ('zscore_grading_wfh_recorded_in_month', "SUM(chm.zscore_grading_wfh_recorded_in_month)"),
+            ('days_ration_given_child', "SUM(chm.days_ration_given_child)"),
+        )
+        return """
+        INSERT INTO "{tablename}" (
+            {columns}
+        ) (SELECT
+            {calculations}
+            FROM "{ucr_child_monthly_table}" ucr
+            LEFT OUTER JOIN "{child_health_monthly_table}" chm ON ucr.doc_id = chm.case_id AND ucr.month = chm.month AND ucr.awc_id = chm.awc_id
```

Comment:
I'm not sure that it's necessary, but I think I included it because there's an index on `awc_id, case_id` for child health monthly table, so it may have made better use of that index? Though not sure that would matter since it would be loading the entire table anyways